sdk/python/kfp/_client.py

```diff
@@ -1280,11 +1282,25 @@ def upload_pipeline_version(self,
 
         if pipeline_name:
             pipeline_id = self.get_pipeline_id(pipeline_name)
-
-        response = self._upload_api.upload_pipeline_version(
-            pipeline_package_path,
+        kwargs = dict(
             name=pipeline_version_name,
-            pipelineid=pipeline_id)
+            pipelineid=pipeline_id,
+        )
+
+        if description:
+            kwargs['description'] = description
+        try:
+            response = self._upload_api.upload_pipeline_version(
+                pipeline_package_path, **kwargs)
+        except kfp_server_api.exceptions.ApiTypeError as e:
+            # ToDo: Remove this once we drop support for kfp_server_api < 1.7
+            if 'description' in e.message:
```

Comment:
nit: I wonder if there could be a more specific and reliable check.

Reply:
I added a bit more of specific. But it's hard since the arguments accepted are just a local variable of the function generated by OpenAPI spec. Open to alternatives